### PR TITLE
fix: card overlay not toggling between upvote and bookmark

### PR DESCRIPTION
**Pull request by Abhi-Bohora**
## Changes
fixes: [#1510](https://github.com/dailydotdev/daily/issues/1510)

### Describe what this PR does
The problem here was if i click on upvote and click the bookmark afterwards`(upvote should be still clicked)` the overlay of the upvote still remains in the card and the overlay the bookmark doesn't appear.  but the opposite works fine if i click on bookmark first and than on upvote`(bookmark should be still clicked)` then it first shows the overlay of bookmark and than overlay of the upvote when each of the specific button is pressed.

so I have come up with this solution to fix that issue, I don't know if this is the best approach to solve this issue. and  please i need feedback if there can be easier solution or if i am missing something 🙏

Before:

https://github.com/user-attachments/assets/2b7cff35-28ee-434c-b7ca-d83511c73356

After:

https://github.com/user-attachments/assets/518c71b0-3d26-410b-8a29-e4ace737cd6c

## Events

Did you introduce any new tracking events?

If yes please remove the comment HTML comment tags and fill the table below

Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

-->

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### CODE REVIEW

**Comment by Abhi-Bohora** [comment]
things i have noticed while solving this issue is if you click on upvote and than click on bookmarks afterwards than this  if condition 
```tsx
if (shouldShowOverlay && onShare) {
  return (
    <CardCoverShare
      post={post}
      onCopy={onInteract}
      onShare={() => {
        onInteract();
        onShare(post);
      }}
    />
  );
}
```
 gets triggered both the time. while going otherway if you first click on bookmark and than upvote than each individual if condition gets triggered properly and works fine.

File: packages/shared/src/hooks/feed/useCardCover.tsx [hunk 1]
```diff
@@ -28,9 +28,21 @@ export const useCardCover = ({
...
+    if (shouldShowOverlay) {
+      setInteraction('upvote');
+    }
+  }, [shouldShowOverlay]);
+
+  useEffect(() => {
+    if (shouldShowReminder) {
+      setInteraction('bookmark');
+    }
+  }, [shouldShowReminder]);
```
Comment (sshanzel) [inline packages/shared/src/hooks/feed/useCardCover.tsx:42]: Looks amazing to see this get fixed. Although not a big fan of use state/use effect combinations. Can we maybe check within the `usePostShareLoop` if we can incorporate the changes there about tracking the current interaction?

**Comment by Abhi-Bohora** [comment]
Hello @sshanzel thanks for feedback, i have implemented this way tracking the current interaction and showing the overlay accordingly. please have a look and if you have something in mind please suggest me i will try to implement that way. currently this is what i came up with.

**Review by sshanzel — APPROVED** [review]
Amazing effort. One last thing then I think we can start testing and see if we can merge. Again, wanted to thank you for all your contribution 🙏

File: packages/shared/src/hooks/feed/useCardCover.tsx [hunk 2]
```diff
@@ -26,11 +26,12 @@ export const useCardCover = ({
   onShare,
   className = {},
 }: UseCardCoverProps): UseCardCover => {
-  const { shouldShowOverlay, onInteract } = usePostShareLoop(post);
+  const { shouldShowOverlay, onInteract, currentInteraction } =
+    usePostShareLoop(post);
   const shouldShowReminder = useBookmarkReminderCover(post);
```
Comment (sshanzel) [inline packages/shared/src/hooks/feed/useCardCover.tsx:31]: We can now remove this line and have it coming from the hook.

Comment (sshanzel) [reply packages/shared/src/hooks/feed/useCardCover.tsx:31]: Sorry, this exact line 31. Then from the hook, we return `shouldShowReminder` from inside of it and be used here.

Comment (Abhi-Bohora) [reply packages/shared/src/hooks/feed/useCardCover.tsx]: got it.

**Review by sshanzel — APPROVED** [review]
Tested locally, and it worked wonderfully! 🤩 

Thank you for your patience with all the changes requested 🙏

**Pull request merged**